### DISCUSSION TITLE
Problems with agents scales

### DISCUSSION THREAD

**Discussion by daniele-baccega** [FLAMEGPU/FLAMEGPU2#1167] [Q&A]
Hello,
I'm having serious problems with agents scales. In the model I have different types of agents that are rooms with the same height, with different lengths and widths and a different rotation (0, 90, 180, 270 degrees).
I tried to use for example `room_agt.setScaleYVariable(HEIGHT_OBJ);`, where HEIGHT_OBJ is the variable that represents the height of the room, but in the visualisation I obtained rooms with different heights.

Moreover, I also tried to use `room_agt.setModelScale(1.0f);` , `room_agt.setScaleXVariable(LENGTH_OBJ);`, `room_agt.setScaleZVariable(WIDTH_OBJ);`, and `room_agt.setYawVariable(YAW);`, but in this way some rooms were aggregated in near (0, 0, 0) with a wrong yaw.

I'm not understanding how the scales work. My obj has already the right dimensions (e.g. height 3, length 5 and width 4).

Is there a limit in the number of agents types? Because here we have two 'floors', with one 'floor', it works.

Here you can find two screenshot:
![Screenshot from 2024-01-02 14-00-12](https://github.com/FLAMEGPU/FLAMEGPU2/assets/94167431/fde5a8aa-6c71-4f27-8f96-5d31f7b43963)

![Screenshot from 2024-01-02 14-00-24](https://github.com/FLAMEGPU/FLAMEGPU2/assets/94167431/f03712f0-bbd2-404d-b24d-6af536920a98)

Thank you!

**Comment by Robadob** [comment]
Scales are relative scaling of the original model size (so 1.0 provides
default scaling). So if you want absolute scaling you would want the input
model to have dimensions 1x1x1, I think this is the case for the primitives
(cube/sphere) provided by FLAMEGPU.


(I'm currently travelling, won't be able to actually test that it works as
designed till later in the week)

**Reply by daniele-baccega** [reply]
I solved the problem simplifying the agents types: now I have only two types, before I had around 100. Now it works!

Thanks!
